lib/ast/definition.cpp

```diff
@@ -73,12 +73,30 @@ void kore_definition::add_module(sptr<kore_module> module) {
 void kore_definition::insert_reserved_symbols() {
   auto mod = kore_module::create("K-RAW-TERM");
   auto decl = kore_symbol_declaration::create("rawTerm", true);
-  auto sort = kore_composite_sort::create("SortKItem");
+  auto kitem = kore_composite_sort::create("SortKItem");
+  std::shared_ptr<kore_composite_sort> sort;
 
-  decl->get_symbol()->add_sort(sort);
-  decl->get_symbol()->add_argument(sort);
+  decl->get_symbol()->add_sort(kitem);
+  decl->get_symbol()->add_argument(kitem);
   mod->add_declaration(std::move(decl));
 
+  for (auto const &cat : hooked_sorts_) {
+    switch (cat.first.cat) {
+    case sort_category::Map:
+    case sort_category::List:
+    case sort_category::Set:
+    case sort_category::RangeMap:
+      decl = kore_symbol_declaration::create(
+          "rawCollection_" + std::to_string((int)cat.first.cat), true);
```

Comment:
As we're using the same `rawCollection_i` schema elsewhere, let's pull the computation of the symbol name for each category out into a function.